### ISSUE TITLE
Add a11y identifier to BottomSheetViewModel

### ISSUE THREAD

**Issue by data-sync-user** [mozilla-mobile/firefox-ios#20019]
**Background**
Currently, we are missing an a11y identifier for the BottomSheetViewModel, we should add this identifier so that it can be set when initializing the view model (similar to the accessibility label). It should model how our other components (i.e. `LinkButtonViewModel`) are passing in the a11y identifier. 

Reference: This ticket was created off the work done for the close button here: [https://github.com/mozilla-mobile/firefox-ios/pull/19848](https://github.com/mozilla-mobile/firefox-ios/pull/19848|smart-link) 

**Acceptance Criteria:**

We should ensure that the `BottomSheetViewModel` also passes in an a11y identifier for the close button in the Component Library. 

Any views in our `Client` project that uses the `BottomSheetViewModel` should also have an a11y identifier.

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-9052)
